src/hddsupport.c

```diff
@@ -93,16 +93,63 @@ static int hddCheckHDProKit(void)
 #define PFS_ZONE_SIZE 8192
 #define PFS_FRAGMENT 0x00000000
 
-static int CreateOPLPartition(const char *oplPart, const char *mountpoint)
+static int hddFindOPLPartition(void)
+{
+    static config_set_t *config;
+    char name[64];
+    int fd, ret = 0;
+
+    fileXioUmount(hddPrefix);
+
+    ret = fileXioMount(hddPrefix, "hdd0:__common", FIO_MT_RDWR);
+    if (ret == 0) {
+        fd = open("pfs0:conf_hdd.cfg", O_RDONLY);
+        if (fd >= 0) {
+            config = configAlloc(0, NULL, "pfs0:conf_hdd.cfg");
+            configRead(config);
+
+            configGetStrCopy(config, "hdd_partition", name, sizeof(name));
+            snprintf(oplPart, sizeof(oplPart), "hdd0:%s", name);
+
+            configFree(config);
+            close(fd);
+
+            if (!strcmp(name, "__common"))
+                return 0;
+
+            fileXioUmount(hddPrefix);
+            return 1;
+        }
+
+        fd = open("pfs0:conf_hdd.cfg", O_CREAT | O_TRUNC | O_WRONLY);
+        if (fd >= 0) {
+            config = configAlloc(0, NULL, "pfs0:conf_hdd.cfg");
+            configRead(config);
+
+            configSetStr(config, "hdd_partition", "+OPL");
+            configWrite(config);
+
+            configFree(config);
+            close(fd);
+        }
+    }
+
+    snprintf(oplPart, sizeof(oplPart), "hdd0:+OPL");
+    fileXioUmount(hddPrefix);
+
+    return 1;
+}
+
+static int hddCreateOPLPartition(const char *name)
 {
     int formatArg[3] = {PFS_ZONE_SIZE, 0x2d66, PFS_FRAGMENT};
     int fd, result;
     char cmd[43];
 
-    sprintf(cmd, "%s,,,128M,PFS", oplPart);
+    sprintf(cmd, "%s,,,128M,PFS", name);
```

Comment:
There is 2 warnings regarding this line:
```gcc
src/hddsupport.c: In function 'hddLoadModules.part.0':
src/hddsupport.c:149:19: warning: '%s' directive writing up to 127 bytes into a region of size 43 [-Wformat-overflow=]
  149 |     sprintf(cmd, "%s,,,128M,PFS", name);
      |                   ^~
src/hddsupport.c:149:5: note: 'sprintf' output between 12 and 139 bytes into a destination of size 43
  149 |     sprintf(cmd, "%s,,,128M,PFS", name);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
` char cmd[140];` wil be enough

Reply:
I get a heap of warnings like this since gcc was updated, because the return value of snprintf is never checked for truncation. I’ll change this one though thanks.

Reply:
Yeah, I thought about fixing at least part of the warnings, maybe someday I will do it :D